### PR TITLE
Test descriptor estimations

### PR DESCRIPTION
**Pull request by Zeskbest**

### CODE REVIEW

File: tests/test_estimate_descriptor.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        for planVersion in range(min(EDVa) - 10, max(EDVa) + 10):
+            with self.subTest(descriptor_version=planVersion):
+                try:
+                    self.faceEngine.createFaceDescriptorEstimator(descriptorVersion=planVersion)
+                except RuntimeError as e:
+                    if planVersion in EDVa:
+                        raise AssertionError(f"Descriptor version {planVersion} is not supported. But must be.") from e
+                else:
+                    if planVersion not in EDVa:
+                        raise AssertionError(f"Descriptor version {planVersion} is supported. But should not.")
```
Comment (matemax) [inline tests/test_estimate_descriptor.py]: other test

File: tests/test_estimate_descriptor.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        for planVersion in EDVa:
+            extractor = self.faceEngine.createFaceDescriptorEstimator(descriptorVersion=planVersion)
+            for descriptorVersion in set(EDVa) - {planVersion}:
+                descriptorOfAnotherVersion = self.faceEngine.createFaceDescriptorFactory(
+                    descriptorVersion=descriptorVersion
+                ).generateDescriptor()
+                with self.subTest(planVersion=planVersion, descriptorVersion=descriptorVersion):
+                    print(f"Plan {planVersion}, empty descriptor version {descriptorVersion}")
+                    descriptor = extractor.estimate(warp, descriptor=descriptorOfAnotherVersion)
+                    self.assert_descriptor(planVersion, descriptor)
```
Comment (matemax) [inline tests/test_estimate_descriptor.py]: + batch estimation tests

**Review by matemax — APPROVED** [review]

File: tests/test_estimate_descriptor.py (hunk 2 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py]: done

File: tests/test_estimate_descriptor.py (hunk 1 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py]: okay

File: tests/test_estimate_descriptor.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        """
+        Test correctly estimate descriptor.
+        """
+        for planVersion in EDVa:
+            for kw in (
+                dict(),
+                dict(descriptor=self.faceEngine.createFaceDescriptorFactory(planVersion).generateDescriptor()),
+            ):
+                extractor = self.faceEngine.createFaceDescriptorEstimator(descriptorVersion=planVersion)
+                with self.subTest(planVersion=planVersion, external_descriptor=bool(kw)):
```
Comment (matemax) [inline tests/test_estimate_descriptor.py]: `external_descriptor` -case

File: tests/test_estimate_descriptor.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    @staticmethod
+    def assert_descriptor(expectedVersion: int, descriptor: FaceDescriptor) -> None:
+        """
+        Assert extracted descriptor.
+
+        Args:
+            expectedVersion: expected descriptor version
+            descriptor: extracted descriptor
+        """
```
Comment (matemax) [inline tests/test_estimate_descriptor.py:118]: + assert type

**Review by matemax — APPROVED** [review]

File: tests/test_estimate_descriptor.py (hunk 3 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py]: camel case everywhere

Comment (matemax) [reply tests/test_estimate_descriptor.py]: `planVersion=planVersion, external_descriptor=bool(kw)`
realy?

File: lunavl/sdk/faceengine/descriptors.py [hunk 5]
```diff
@@ -157,9 +168,23 @@ def append(self, descriptor: FaceDescriptor) -> None:
...
         self.scores.append(descriptor.garbageScore)
 
+    def __repr__(self) -> str:
+        """
+        Representation.
+
+        Returns:
+            str(self.asDict())
+        """
+        full_descriptors = self.asDict()
```
Comment (matemax) [inline lunavl/sdk/faceengine/descriptors.py]: full_descriptors - case

**Review by matemax — APPROVED** [review]

File: tests/test_estimate_descriptor.py (hunk 4 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py:118]: ok

File: tests/test_estimate_descriptor.py (hunk 3 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py]: Won't change. It is easier to read snake case in test logs.

File: tests/test_estimate_descriptor.py (hunk 4 above)
Comment (Zeskbest) [reply tests/test_estimate_descriptor.py:118]: Type asserted.

File: lunavl/sdk/faceengine/descriptors.py (hunk 5 above)
Comment (Zeskbest) [reply lunavl/sdk/faceengine/descriptors.py]: done

**Pull request merged**